### PR TITLE
Add GINConv support to HeteroGraphConv

### PR DESCRIPTION
**Pull request by rbSparky**
Covers Issue #311 

This operation needs to be done differently for hetero graphs
```
    if g isa GNNHeteroGraph
        #
    else
        l.nn((1 + ofeltype(x, l.ϵ)) * x + m)
    end
```

[EDIT: done]

### CODE REVIEW

**Comment by rbSparky** [comment]
ready for merge?

File: src/layers/conv.jl [hunk 1]
```diff
@@ -680,10 +680,17 @@ Flux.trainable(l::GINConv) = (nn = l.nn,)
...
-    l.nn((1 + ofeltype(x, l.ϵ)) * x + m)
+    xj, xi = expand_srcdst(g, x) 
+ 
+    m = propagate(copy_xj, g, l.aggr, xj = xj)
+    
+    if g isa GNNHeteroGraph
+        l.nn((1 .+ ofeltype(xi, l.ϵ)) .* xi .+ m)
+    else
+        l.nn((1 + ofeltype(x, l.ϵ)) * x + m)
+    end
```
Comment (CarloLucibello) [inline src/layers/conv.jl:693]: why there is an `if` here instead of just using `xi`?

Comment (rbSparky) [reply src/layers/conv.jl]: fixed

**Comment by rbSparky** [comment]
Changes done

**Pull request merged**